client/scripts/views/pages/view_proposal/body.ts

```diff
@@ -338,12 +336,38 @@ export const ProposalBodySaveEdit: m.Component<{
           e.preventDefault();
           parentState.saving = true;
           parentState.quillEditorState.editor.enable(false);
-          const itemText = parentState.quillEditorState.markdownMode
-            ? parentState.quillEditorState.editor.getText()
-            : JSON.stringify(parentState.quillEditorState.editor.getContents());
+          const { quillEditorState } = parentState;
+          const itemText = quillEditorState.markdownMode
+            ? quillEditorState.editor.getText()
+            : JSON.stringify(quillEditorState.editor.getContents());
+          let mentions;
+          if (isThread) {
+            const currentDraftMentions = !quillEditorState
+              ? []
+              : quillEditorState.markdownMode
+                ? parseMentionsForServer(quillEditorState.editor.getText(), true)
+                : parseMentionsForServer(quillEditorState.editor.getContents(), false);
+            const previousDraft = JSON.parse((item as OffchainThread).versionHistory[0]);
```

Comment:
This will break if the old versionHistory is not valid JSON.

Reply:
No longer an issue with new VersionHistory system, which parses all versionHistories while modeling from server!